docker/cross.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC1091
+
+set -x
+set -euo pipefail
+
+. lib.sh
+
+main() {
+    install_packages ca-certificates curl gcc libc6-dev
+
+    # since we can't share outside the current docker context,
+    # we need to install from git, rather than from path
+    curl --proto "=https" --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+    source "$HOME"/.cargo/env
+    cargo install cross --git https://github.com/cross-rs/cross --locked
```

Comment:
I somewhat avoided that, because it would require the script to be sufficiently distant from `docker buildx`, but I guess we could do it. It would require a release build of `cross` inside `build_docker_image`, and then make the binaries (both `cross` and `cross-util`) available to the Dockerfile context (either by making it the project directory, or via copying), and then the build can start. I wasn't sure if that was the best approach.

Reply:
We could, somewhat easier, make the parent directory the Dockerfile context just for this image, build it inside the container via a script via a temporary layer, and then copy the built binary `using --from`.